pkg/config/libpodConfig.go

```diff
@@ -0,0 +1,316 @@
+	configs, err := systemLibpodConfigs()
+	if err != nil {
+		return nil, errors.Wrapf(err, "error finding config on system")
+	}
+
+	for _, path := range configs {
+		config, err = readLibpodConfigFromFile(path, config)
+		if err != nil {
+			return nil, errors.Wrapf(err, "error reading system config %q", path)
+		}
+	}
+
+	// Since runc does not currently support cgroupV2
+	// Change to default crun on first running of libpod.conf
+	// TODO Once runc has support for cgroups, this function should be removed.
+	if !config.CgroupCheck && unshare.IsRootless() {
+		cgroupsV2, err := isCgroup2UnifiedMode()
+		if err != nil {
+			return nil, err
+		}
+		if cgroupsV2 {
+			path, err := exec.LookPath("crun")
+			if err != nil {
+				// Can't find crun path so do nothing
+				logrus.Warnf("Can not find crun package on the host, containers might fail to run on cgroup V2 systems without crun: %q", err)
+			} else {
+				config.CgroupCheck = true
+				config.OCIRuntime = path
+			}
+		}
+	}
+
+	// Relative paths can cause nasty bugs, because core paths we use could
+	// shift between runs (or even parts of the program - the OCI runtime
+	// uses a different working directory than we do, for example.
+	if !filepath.IsAbs(config.StaticDir) {
+		return nil, errors.Wrapf(ErrInvalidArg, "static directory must be an absolute path - instead got %q", config.StaticDir)
+	}
+	if !filepath.IsAbs(config.TmpDir) {
+		return nil, errors.Wrapf(ErrInvalidArg, "temporary directory must be an absolute path - instead got %q", config.TmpDir)
+	}
+	if !filepath.IsAbs(config.VolumePath) {
+		return nil, errors.Wrapf(ErrInvalidArg, "volume path must be an absolute path - instead got %q", config.VolumePath)
+	}
+
+	return config, nil
+}
+
+// readConfigFromFile reads the specified config file at `path` and attempts to
+// unmarshal its content into a Config. The config param specifies the previous
+// default config. If the path, only specifies a few fields in the Toml file
+// the defaults from the config parameter will be used for all other fields.
+func readLibpodConfigFromFile(path string, config *ConfigFromLibpod) (*ConfigFromLibpod, error) {
+	logrus.Debugf("Reading configuration file %q", path)
+	_, err := toml.DecodeFile(path, config)
+	if err != nil {
+		return nil, fmt.Errorf("unable to decode configuration %v: %v", path, err)
+	}
+
+	// For the sake of backwards compat we need to check if the config fields
+	// with *Set suffix are set in the config.  Note that the storage-related
+	// fields are NOT set in the config here but in the storage.conf OR directly
+	// by the user.
+	if config.VolumePath != "" {
+		config.VolumePathSet = true
+	}
+	if config.StaticDir != "" {
+		config.StaticDirSet = true
+	}
+	if config.TmpDir != "" {
+		config.TmpDirSet = true
+	}
+
+	return config, err
+}
+
+func systemLibpodConfigs() ([]string, error) {
+	if unshare.IsRootless() {
+		path, err := rootlessLibpodConfigPath()
```

Comment:
Warning here, about the user should have used
$HOME/.config/containers/containers.conf